lib/element.ts

```diff
@@ -751,14 +749,16 @@ export class ElementFinder extends WebdriverWebElement {
        * of
        *     evaluating fn.
        */
-      this.then = (fn: Function, errorFn: Function) => {
-        return this.elementArrayFinder_.then((actionResults: any) => {
-          if (!fn) {
-            return actionResults[0];
-          }
-          return fn(actionResults[0]);
-        }, errorFn);
-      };
+      this.then =
```

Comment:
I don't think this is a good case for generics, because we don't actually know what value it will return based on the arguments to the function - it's based on what has been called in the chain.